shared/engine/index.js

```diff
@@ -192,7 +192,9 @@ class Engine {
       }
     }
 
-    this.sessionIDToResponse[sessionID] = wrappedResponse
+    if (sessionID) {
```

Comment:
there being no sessionid is a problem i think, but we should just log that here and not add it to the map